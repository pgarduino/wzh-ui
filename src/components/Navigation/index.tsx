import React, {FC} from 'react';

import {
    Route,
    Link,
    Redirect,
} from 'react-router-dom';

import { Users } from '../../pages/Users';
import { UserView } from '../../pages/UserView';
import { Tasks } from '../../pages/Tasks';

import './index.css';

import { Layout, Menu } from 'antd';
import { BranchesOutlined, UserOutlined } from '@ant-design/icons';

const { Header, Content, Footer, Sider } = Layout;


const Navigation: FC = () => {
    return(
        <Layout>
            <Sider
                breakpoint="lg"
                collapsedWidth="0"
            >
                <div className="logo" />
                <Menu theme="dark" mode="inline" defaultSelectedKeys={['1']}>
                    <Menu.Item key="1" icon={<UserOutlined />}>
                        <Link to="/users">Users</Link>
                    </Menu.Item>
                    <Menu.Item key="2" icon={<BranchesOutlined />}>
                        <Link to="/tasks">Tasks</Link>
                    </Menu.Item>
                </Menu>
            </Sider>
            <Layout>
                <Header className="site-layout-sub-header-background" style={{ padding: 0 }} />
                <Content style={{ margin: '24px 16px 0' }}>
                    <div className="site-layout-background" style={{ padding: 24, minHeight: 360 }}>
                        <Route path="/users/:id" component={UserView} />
                        <Route exact path="/">
                            <Redirect to="/users" />
                        </Route>
                        <Route exact path="/users" component={Users} />
                        <Route exact path="/tasks" component={Tasks} />
                    </div>
                </Content>
                <Footer style={{ textAlign: 'center' }}>Â© Created by Wzh</Footer>
            </Layout>
        </Layout>
    );
};

export default Navigation;