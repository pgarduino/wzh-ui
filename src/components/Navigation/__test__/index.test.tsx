import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Navigation from '..';

describe("Navigation", () => {
    beforeAll(() => {
        Object.defineProperty(window, "matchMedia", {
            writable: true,
            value: jest.fn().mockImplementation(query => ({
                matches: false,
                media: query,
                onchange: null,
                addListener: jest.fn(), // Deprecated
                removeListener: jest.fn(), // Deprecated
                addEventListener: jest.fn(),
                removeEventListener: jest.fn(),
                dispatchEvent: jest.fn(),
            }))
        });
    });
    test('Should render', () => {
        const wrapper = render(
            <BrowserRouter> <Navigation /> </BrowserRouter>
        )

        expect(wrapper).toMatchSnapshot()
    });
})