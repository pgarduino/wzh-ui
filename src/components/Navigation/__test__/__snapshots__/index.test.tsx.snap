// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Should render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
       
      <section
        class="ant-layout ant-layout-has-sider"
      >
        <aside
          class="ant-layout-sider ant-layout-sider-dark"
          style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
        >
          <div
            class="ant-layout-sider-children"
          >
            <div
              class="logo"
            />
            <ul
              class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark"
              data-menu-list="true"
              role="menu"
              tabindex="0"
            >
              <li
                class="ant-menu-item ant-menu-item-selected"
                data-menu-id="rc-menu-uuid-test-1"
                role="menuitem"
                style="padding-left: 24px;"
                tabindex="-1"
              >
                <span
                  aria-label="user"
                  class="anticon anticon-user ant-menu-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="user"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                    />
                  </svg>
                </span>
                <span
                  class="ant-menu-title-content"
                >
                  <a
                    href="/users"
                  >
                    Users
                  </a>
                </span>
              </li>
              <li
                class="ant-menu-item"
                data-menu-id="rc-menu-uuid-test-2"
                role="menuitem"
                style="padding-left: 24px;"
                tabindex="-1"
              >
                <span
                  aria-label="branches"
                  class="anticon anticon-branches ant-menu-item-icon"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="branches"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="64 64 896 896"
                    width="1em"
                  >
                    <path
                      d="M740 161c-61.8 0-112 50.2-112 112 0 50.1 33.1 92.6 78.5 106.9v95.9L320 602.4V318.1c44.2-15 76-56.9 76-106.1 0-61.8-50.2-112-112-112s-112 50.2-112 112c0 49.2 31.8 91 76 106.1V706c-44.2 15-76 56.9-76 106.1 0 61.8 50.2 112 112 112s112-50.2 112-112c0-49.2-31.8-91-76-106.1v-27.8l423.5-138.7a50.52 50.52 0 0034.9-48.2V378.2c42.9-15.8 73.6-57 73.6-105.2 0-61.8-50.2-112-112-112zm-504 51a48.01 48.01 0 0196 0 48.01 48.01 0 01-96 0zm96 600a48.01 48.01 0 01-96 0 48.01 48.01 0 0196 0zm408-491a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                    />
                  </svg>
                </span>
                <span
                  class="ant-menu-title-content"
                >
                  <a
                    href="/tasks"
                  >
                    Tasks
                  </a>
                </span>
              </li>
            </ul>
            <div
              aria-hidden="true"
              style="display: none;"
            />
          </div>
        </aside>
        <section
          class="ant-layout"
        >
          <header
            class="ant-layout-header site-layout-sub-header-background"
            style="padding: 0px;"
          />
          <main
            class="ant-layout-content"
            style="margin: 24px 16px 0px;"
          >
            <div
              class="site-layout-background"
              style="padding: 24px; min-height: 360px;"
            >
              <div>
                <div
                  class="ant-table-wrapper"
                >
                  <div
                    class="ant-spin-nested-loading"
                  >
                    <div
                      class="ant-spin-container"
                    >
                      <div
                        class="ant-table ant-table-empty"
                      >
                        <div
                          class="ant-table-container"
                        >
                          <div
                            class="ant-table-content"
                          >
                            <table
                              style="table-layout: auto;"
                            >
                              <colgroup />
                              <thead
                                class="ant-table-thead"
                              >
                                <tr>
                                  <th
                                    class="ant-table-cell"
                                  >
                                    id
                                  </th>
                                  <th
                                    class="ant-table-cell"
                                  >
                                    Name
                                  </th>
                                  <th
                                    class="ant-table-cell"
                                  >
                                    Username
                                  </th>
                                  <th
                                    class="ant-table-cell"
                                  >
                                    Email
                                  </th>
                                  <th
                                    class="ant-table-cell"
                                  >
                                    Action
                                  </th>
                                </tr>
                              </thead>
                              <tbody
                                class="ant-table-tbody"
                              >
                                <tr
                                  class="ant-table-placeholder"
                                >
                                  <td
                                    class="ant-table-cell"
                                    colspan="5"
                                  >
                                    <div
                                      class="ant-empty ant-empty-normal"
                                    >
                                      <div
                                        class="ant-empty-image"
                                      >
                                        <svg
                                          class="ant-empty-img-simple"
                                          height="41"
                                          viewBox="0 0 64 41"
                                          width="64"
                                          xmlns="http://www.w3.org/2000/svg"
                                        >
                                          <g
                                            fill="none"
                                            fill-rule="evenodd"
                                            transform="translate(0 1)"
                                          >
                                            <ellipse
                                              class="ant-empty-img-simple-ellipse"
                                              cx="32"
                                              cy="33"
                                              rx="32"
                                              ry="7"
                                            />
                                            <g
                                              class="ant-empty-img-simple-g"
                                              fill-rule="nonzero"
                                            >
                                              <path
                                                d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                              />
                                              <path
                                                class="ant-empty-img-simple-path"
                                                d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                              />
                                            </g>
                                          </g>
                                        </svg>
                                      </div>
                                      <div
                                        class="ant-empty-description"
                                      >
                                        No Data
                                      </div>
                                    </div>
                                  </td>
                                </tr>
                              </tbody>
                            </table>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </main>
          <footer
            class="ant-layout-footer"
            style="text-align: center;"
          >
            © Created by Wzh
          </footer>
        </section>
      </section>
       
    </div>
  </body>,
  "container": <div>
     
    <section
      class="ant-layout ant-layout-has-sider"
    >
      <aside
        class="ant-layout-sider ant-layout-sider-dark"
        style="flex: 0 0 200px; max-width: 200px; min-width: 200px; width: 200px;"
      >
        <div
          class="ant-layout-sider-children"
        >
          <div
            class="logo"
          />
          <ul
            class="ant-menu ant-menu-root ant-menu-inline ant-menu-dark"
            data-menu-list="true"
            role="menu"
            tabindex="0"
          >
            <li
              class="ant-menu-item ant-menu-item-selected"
              data-menu-id="rc-menu-uuid-test-1"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="user"
                class="anticon anticon-user ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="user"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  href="/users"
                >
                  Users
                </a>
              </span>
            </li>
            <li
              class="ant-menu-item"
              data-menu-id="rc-menu-uuid-test-2"
              role="menuitem"
              style="padding-left: 24px;"
              tabindex="-1"
            >
              <span
                aria-label="branches"
                class="anticon anticon-branches ant-menu-item-icon"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="branches"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M740 161c-61.8 0-112 50.2-112 112 0 50.1 33.1 92.6 78.5 106.9v95.9L320 602.4V318.1c44.2-15 76-56.9 76-106.1 0-61.8-50.2-112-112-112s-112 50.2-112 112c0 49.2 31.8 91 76 106.1V706c-44.2 15-76 56.9-76 106.1 0 61.8 50.2 112 112 112s112-50.2 112-112c0-49.2-31.8-91-76-106.1v-27.8l423.5-138.7a50.52 50.52 0 0034.9-48.2V378.2c42.9-15.8 73.6-57 73.6-105.2 0-61.8-50.2-112-112-112zm-504 51a48.01 48.01 0 0196 0 48.01 48.01 0 01-96 0zm96 600a48.01 48.01 0 01-96 0 48.01 48.01 0 0196 0zm408-491a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                  />
                </svg>
              </span>
              <span
                class="ant-menu-title-content"
              >
                <a
                  href="/tasks"
                >
                  Tasks
                </a>
              </span>
            </li>
          </ul>
          <div
            aria-hidden="true"
            style="display: none;"
          />
        </div>
      </aside>
      <section
        class="ant-layout"
      >
        <header
          class="ant-layout-header site-layout-sub-header-background"
          style="padding: 0px;"
        />
        <main
          class="ant-layout-content"
          style="margin: 24px 16px 0px;"
        >
          <div
            class="site-layout-background"
            style="padding: 24px; min-height: 360px;"
          >
            <div>
              <div
                class="ant-table-wrapper"
              >
                <div
                  class="ant-spin-nested-loading"
                >
                  <div
                    class="ant-spin-container"
                  >
                    <div
                      class="ant-table ant-table-empty"
                    >
                      <div
                        class="ant-table-container"
                      >
                        <div
                          class="ant-table-content"
                        >
                          <table
                            style="table-layout: auto;"
                          >
                            <colgroup />
                            <thead
                              class="ant-table-thead"
                            >
                              <tr>
                                <th
                                  class="ant-table-cell"
                                >
                                  id
                                </th>
                                <th
                                  class="ant-table-cell"
                                >
                                  Name
                                </th>
                                <th
                                  class="ant-table-cell"
                                >
                                  Username
                                </th>
                                <th
                                  class="ant-table-cell"
                                >
                                  Email
                                </th>
                                <th
                                  class="ant-table-cell"
                                >
                                  Action
                                </th>
                              </tr>
                            </thead>
                            <tbody
                              class="ant-table-tbody"
                            >
                              <tr
                                class="ant-table-placeholder"
                              >
                                <td
                                  class="ant-table-cell"
                                  colspan="5"
                                >
                                  <div
                                    class="ant-empty ant-empty-normal"
                                  >
                                    <div
                                      class="ant-empty-image"
                                    >
                                      <svg
                                        class="ant-empty-img-simple"
                                        height="41"
                                        viewBox="0 0 64 41"
                                        width="64"
                                        xmlns="http://www.w3.org/2000/svg"
                                      >
                                        <g
                                          fill="none"
                                          fill-rule="evenodd"
                                          transform="translate(0 1)"
                                        >
                                          <ellipse
                                            class="ant-empty-img-simple-ellipse"
                                            cx="32"
                                            cy="33"
                                            rx="32"
                                            ry="7"
                                          />
                                          <g
                                            class="ant-empty-img-simple-g"
                                            fill-rule="nonzero"
                                          >
                                            <path
                                              d="M55 12.76L44.854 1.258C44.367.474 43.656 0 42.907 0H21.093c-.749 0-1.46.474-1.947 1.257L9 12.761V22h46v-9.24z"
                                            />
                                            <path
                                              class="ant-empty-img-simple-path"
                                              d="M41.613 15.931c0-1.605.994-2.93 2.227-2.931H55v18.137C55 33.26 53.68 35 52.05 35h-40.1C10.32 35 9 33.259 9 31.137V13h11.16c1.233 0 2.227 1.323 2.227 2.928v.022c0 1.605 1.005 2.901 2.237 2.901h14.752c1.232 0 2.237-1.308 2.237-2.913v-.007z"
                                            />
                                          </g>
                                        </g>
                                      </svg>
                                    </div>
                                    <div
                                      class="ant-empty-description"
                                    >
                                      No Data
                                    </div>
                                  </div>
                                </td>
                              </tr>
                            </tbody>
                          </table>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </main>
        <footer
          class="ant-layout-footer"
          style="text-align: center;"
        >
          © Created by Wzh
        </footer>
      </section>
    </section>
     
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
